### DISCUSSION TITLE
ArgoCD Release Template

### DISCUSSION THREAD

**Discussion by adecchi-2inno** [helmfile/helmfile#448] [Questions]
I have setup a release template for Argo Application and then call it at each release, like the following example:

`  
   - name: grafana
    <<: *argocd-apps
    namespace: monitoring
    createNamespace: true
    disableValidation: true
    missingFileHandler: Warn
    labels:
      release-name: "{{ `{{ .Release.Name }}` }}"
      layer: "infra"
    set:
      - name: source.helm.file
        file: "helmfiles/grafana/{{ .Environment.Name }}/values.yaml.gotmpl"
`

So, my doubt here is to try to understand if possible to template a values.yaml.gotmpl used at **set** section previous to submit it to ArgoCD
My intention is to template the values.yaml used by the grafana helm-chart and create the Grafana Argo CD Application. My intention is to use environment variable(**{{ requiredEnv "ARGOCD_INGRESS_NAME" }}**) to then pass it the gotmpl to create the Grafana ingress. If not possible , is possible to use [envsubst](envsubst (https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) )  or another better way to work with it ?

**Comment by yxxhero** [comment]
@adecchi-2inno why not use `values`?
